src/protocols/bdx/BdxMessages.h

```diff
@@ -0,0 +1,240 @@
+     *  Returns the size of buffer needed to write this message.
+     */
+    size_t PackedSize() const;
+
+    /**
+     * @brief
+     *  Equality check method.
+     */
+    bool operator==(const SendAccept &) const;
+
+    // Transfer Control (required, only one should be set)
+    BitFlags<uint8_t, TransferControlFlags> mTransferCtlFlags;
+
+    uint8_t mVersion       = 0; ///< The agreed upon version for the transfer (required)
+    uint16_t mMaxBlockSize = 0; ///< Chosen max block size to use in transfer (required)
+
+    // Additional metadata (optional, TLV format)
+    // WARNING: there is no guarantee at any point that this pointer will point to valid memory.
+    // It is up to the caller to ensure that the memory pointed to here has not been freed.
+    uint8_t * mMetadata      = nullptr;
+    uint16_t mMetadataLength = 0;
+};
+
+/**
+ * @class ReceiveAccept
+ *
+ * @brief
+ *   The ReceiveAccept message is used to accept a proposed exchange when the
+ *   receiver is the initiator.
+ */
+struct ReceiveAccept
+{
+    /**
+     * @brief
+     *  Pack (write) the message into a PacketBuffer.
+     *
+     * @param[out] aBuffer a PacketBuffer to use to write the message
+     *
+     * @return CHIP_ERROR Any error that occurs when trying to write to the PacketBuffer
+     */
+    CHECK_RETURN_VALUE
+    CHIP_ERROR Pack(System::PacketBuffer & aBuffer) const;
+
+    /**
+     * @brief
+     *  Parse data from an PacketBuffer into a struct instance
+     *
+     *  Note that this struct will store pointers that point into the passed PacketBuffer,
+     *  so it is essential that the PacketBuffer is not modified or freed until after this
+     *  struct is no longer needed.
+     *
+     * @param[in] aBuffer Pointer to a PacketBuffer containing the data
+     *
+     * @return CHIP_ERROR Any error that occurs when trying to read the message
+     */
+    CHECK_RETURN_VALUE
+    CHIP_ERROR Parse(const System::PacketBuffer & aBuffer);
+
+    /**
+     * @brief
+     *  Returns the size of buffer needed to write this message.
+     */
+    size_t PackedSize() const;
+
+    /**
+     * @brief
+     *  Equality check method.
+     */
+    bool operator==(const ReceiveAccept &) const;
+
+    // Transfer Control (required, only one should be set)
+    BitFlags<uint8_t, TransferControlFlags> mTransferCtlFlags;
+
+    // All required
+    uint8_t mVersion       = 0; ///< The agreed upon version for the transfer
+    uint16_t mMaxBlockSize = 0; ///< Chosen max block size to use in transfer
+    uint64_t mStartOffset  = 0; ///< Chosen start offset of data. 0 for no offset.
+    uint64_t mLength       = 0; ///< Length of transfer. 0 if length is indefinite.
+
+    // Additional metadata (optional, TLV format)
+    // WARNING: there is no guarantee at any point that this pointer will point to valid memory.
+    // It is up to the caller to ensure that the memory pointed to here has not been freed.
+    uint8_t * mMetadata      = nullptr;
```

Comment:
Here as well (and elsewhere): all mMetadata pointers ought to be const.

Also, if we are using straight-up public access, why are the fields using `m` prefix?

@andreilitvin If we use public access, shouldn't the fields all be without `m` member prefix?

Reply:
I am used to 'structs do not use m' and 'classes use m and do not provide public access'.

I have not seen a code guideline so I would not block a PR on this. My real preference would be 'use classes, everything private, add accessors' (e.g. then we can handle memory lifetime and such).

Reply:
Perhaps we could revisit this also in #3837 ? Switching from struct to class and adding setters/getters seems like a big a change for this PR. But I can remove "m" from the fields.

Reply:
Absolutely, don't want to derail the PR. If given a choice, I would prefer to iterate instead of trying to make a PR perfect. Perfect takes too long.